hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncTableRegionLocatorImpl.java

```diff
@@ -61,8 +62,23 @@ public CompletableFuture<List<HRegionLocation>> getAllRegionLocations() {
         return conn.registry.getMetaRegionLocations()
           .thenApply(locs -> Arrays.asList(locs.getRegionLocations()));
       }
-      return ClientMetaTableAccessor
-        .getTableHRegionLocations(conn.getTable(TableName.META_TABLE_NAME), tableName);
+      CompletableFuture<List<HRegionLocation>> future = new CompletableFuture<>();
+      addListener(ClientMetaTableAccessor.getTableHRegionLocations(conn
+          .getTable(TableName.META_TABLE_NAME), tableName), (locs, error) -> {
+          if (error != null) {
+            future.completeExceptionally(error);
+            return;
+          }
+          try {
+            locs.forEach(loc -> conn.getLocator()
+              .getNonMetaRegionLocator().addLocationToCache(loc));
+          } catch (Exception e) {
+            future.completeExceptionally(e);
+          } finally {
+            future.complete(locs);
```

Comment:
can you move this to right after the `locs.forEach`?

As is, this will result in completing the future twice when an exception is raised. I think CompletableFuture will handle this correctly, but it's more correct and less confusing to future readers to not do that.

Reply:
sorry, that's a stupid mistake..... Thanks for pointing out. I'll fix this.